src/transformers/trainer.py

```diff
@@ -772,8 +772,8 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
         Trainer's init through :obj:`optimizers`, or subclass and override this method (or :obj:`create_optimizer`
         and/or :obj:`create_scheduler`) in a subclass.
         """
-        self.create_optimizer()
-        self.create_scheduler(num_training_steps)
+        self.optimizer = self.create_optimizer()
+        self.lr_scheduler = self.create_scheduler(optimizer=self.optimizer, num_training_steps=num_training_steps)
```

Comment:
the assignment isn't needed, no?

```suggestion
        self.create_optimizer()
        self.create_scheduler(optimizer=self.optimizer, num_training_steps=num_training_steps)
```

I know that now that you changed the API to return the value, it's sort of a mixed bag.

Reply:
The assignments were missing there originally as seen [here](https://github.com/huggingface/transformers/blob/2772d3e79d66925cf4adeaffd8be610f0ab177b6/src/transformers/trainer.py#L770-L771).

Reply:
I think we are saying the same thing. 

That's why I proposed to restore how it was originally, since the proposed change is redundant.

Reply:
My bad, I got confused. I thought I had removed them. Will do on the next push.